rust/template/api.rs

```diff
@@ -629,7 +631,7 @@ pub unsafe extern "C" fn ddlog_apply_updates(prog: *const HDDlog, upds: *const *
     let mut cmds_vec: Vec<Box<record::UpdCmd>> = Vec::with_capacity(n as usize);
 
     for i in 0..n {
-        cmds_vec.push(Box::from_raw(*upds.offset(i as isize)));
+        cmds_vec.push(Box::from_raw(*upds.add(i)));
```

Comment:
Do you know why Clippy insists on this?

Reply:
To avoid casts: https://rust-lang.github.io/rust-clippy/master/index.html#option_map_unit_fn but I'm not exactly sure why casts are bad / how bad they are.